javav2/example_code/acmpca/src/main/java/com/example/acmpca/scenarios/CreatePrivateCertificateAuthorityAD.java

```diff
@@ -0,0 +1,244 @@
+        String csr = GetCertificateAuthorityCsr(rootCAArn, client);
+        String rootCertificateArn = IssueCertificate(rootCAArn, csr, client);
+        String rootCertificate = GetCertificate(rootCertificateArn, rootCAArn, client);
+        ImportCertificateAuthorityCertificate(rootCertificate, rootCAArn, client);
+    }
+
+    private static AcmPcaClient ClientBuilder(String region) {
+        AcmPcaClient client = AcmPcaClient.builder().region(Region.of(region)).build();
+        return client;
+    }
+
+    private static String CreateCertificateAuthority(CertificateAuthorityConfiguration configCA, CertificateAuthorityType CAtype, AcmPcaClient client) {
+        // Create the request object.
+        CreateCertificateAuthorityRequest createCARequest = CreateCertificateAuthorityRequest.builder()
+                .certificateAuthorityConfiguration(configCA)
+                .idempotencyToken("123987")
+                .certificateAuthorityType(CAtype)
+                .build();
+        
+        try {
+          CreateCertificateAuthorityResponse createCAResult = client.createCertificateAuthority(createCARequest);
+          // Retrieve the ARN of the private CA.
+          String rootCAArn = createCAResult.certificateAuthorityArn();
+          System.out.println("Root CA Arn: " + rootCAArn);
+          return rootCAArn;
+        } catch (AcmPcaException ex) {
+          System.err.println("Failed to create CA: " + ex.awsErrorDetails().errorMessage());
+          throw new RuntimeException("Certificate Authority creation failed", ex);
+        } 
+    }
+
+    private static String GetCertificateAuthorityCsr(String rootCAArn, AcmPcaClient client) {
+
+        // Create the CSR request object.
+        GetCertificateAuthorityCsrRequest csrRequest = 
+            GetCertificateAuthorityCsrRequest.builder()
+                .certificateAuthorityArn(rootCAArn)
+                .build();
+
+        // Create waiter to wait on successful creation of the CSR file.
+        try (AcmPcaWaiter waiter = client.waiter()) {
+         waiter.waitUntilCertificateAuthorityCSRCreated(
+             b -> b.certificateAuthorityArn(csrRequest.certificateAuthorityArn()).build());
+        } catch (AcmPcaException ex) {
+         System.err.println(ex.awsErrorDetails().errorMessage());
+        }
+
+        try {
+          GetCertificateAuthorityCsrResponse csrResult = client.getCertificateAuthorityCsr(csrRequest);
+          // Retrieve and display the CSR.
+          String csr = csrResult.csr();
+          System.out.println(csr);
+          return csr;
+        } catch (AcmPcaException ex) {
+          System.err.println("Failed to get CSR: " + ex.awsErrorDetails().errorMessage());
+          throw new RuntimeException("Failed to get Certificate Authority CSR", ex);
+        }
+    }
+
+    private static String IssueCertificate(String rootCAArn, String csr, AcmPcaClient client) {
+
```

Comment:
I see that the original sample has lots of these have unnecesssary lines. Not gonna keep commenting, but you get the idea